OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -313,7 +313,7 @@ public HackyAI(HackyAIInfo info, ActorInitializer init)
 			claimLayer = World.WorldActor.TraitOrDefault<ResourceClaimLayer>();
 			pathfinder = World.WorldActor.Trait<IPathFinder>();
 
-			isEnemyUnit = unit =>
+			IsEnemyUnit = unit =>
 				Player.Stances[unit.Owner] == Stance.Enemy
 					&& !unit.Info.HasTraitInfo<HuskInfo>()
```

Comment:
It might pay to add `!unit.IsDead` and `!unit.Disposed` here (before the info checks), that might prevent bugs and would allow you to simplify line 554 and 622 a bit.